bliss/simulator/prior.py

```diff
@@ -249,6 +272,45 @@ def _draw_truncated_pareto(alpha, min_x, max_x, n_samples) -> Tensor:
         uniform_samples = torch.rand(n_samples) * u_max
         return min_x / (1.0 - uniform_samples) ** (1 / alpha)
 
+    def _compute_flux_ratio(self, obj_fluxes, run, camcol, field, field_dir, gains):
+        """Query SDSS frames to get flux ratios in units of electron count.
+
+        Args:
+            obj_fluxes: tensor of electron counts for a particular SDSS object
+            run: specific scan
+            camcol: scanline within run
+            field: division of scanline
+            field_dir: directory containing frames corresponding to input field
+            gains: list containing band-specific gain values
+
+        Returns:
+            ratios (Tensor): Ratio of electron counts for each light source in field
+            relative to r-band
+        """
+        ratios = torch.zeros(obj_fluxes.size())
+
+        for i, bnd in enumerate("ugriz"):
+            # read frame corresponding to rcf and band
+            frame_path = field_dir / f"frame-{bnd}-{run:06d}-{camcol:d}-{field:04d}.fits"
+            frame = fits.open(frame_path)
+            calibration = frame[1].data  # pylint: disable=maybe-no-member
+            cal_mean = calibration.mean()
+
+            # gain varies across bands
+            bnd_gain = gains[i]
+            r_gain = gains[2]  # hard-code r-gain
+
+            # distribution of nelec_per_nmgy very clustered around mean
+            nelec_per_nmgy_bnd = bnd_gain / cal_mean
+            nelec_per_nmgy_r = r_gain / cal_mean
+            nelec_per_nmgy_rat = nelec_per_nmgy_bnd / nelec_per_nmgy_r
+
+            # (nelec ratio relative to r-band) * (nmgy/nelec ratio relative to r band)
```

Comment:
I think the code is right, but this comment seems backward. We have (nmgy ratio relative to r), so we multiply by (nelec/nmgy ratio) right? Instead of the other way around

Reply:
Yeah I actually had my units wrong the first time around in the first commit for this branch and had to fix it to be in units of electron counts. I forgot to revise the comment, though.